RFS/src/main/java/com/rfs/common/DocumentReindexer.java

```diff
@@ -6,21 +6,42 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.lucene.document.Document;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.instrumentation.annotations.SpanAttribute;
+import io.opentelemetry.instrumentation.annotations.WithSpan;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 import reactor.util.retry.Retry;
 
 
 public class DocumentReindexer {
     private static final Logger logger = LogManager.getLogger(DocumentReindexer.class);
+    private static final LongCounter documentsProcessedCounter;
+    private static final LongCounter batchesProcessedCounter;
+    static {
+        final var meter = GlobalOpenTelemetry.getMeter("DocumentReindexer");
+        documentsProcessedCounter = meter.counterBuilder("documentsProcessed")
+            .setDescription("Counts the number of documents processed")
+            .setUnit("1")
+            .build();
```

Comment:
Curious what we think about this as a pattern - have the metrics defined in the class where they are used vs a centralized store.

Reply:
@gregschohn Thanks for your thoughts, I see two tracks:
- **Where/how metrics are constructed**
  - It doesn't matter to me if they are statically defined or initialized through a DI pattern (lazy providers?)  I'm not married to use of `GlobalOpenTelemetry`.  
- **Where are metrics declared and used**
  - Generally: I typically see two models, single giant metrics file vs just-in-time-declaration.  As someone that has often had to splunk through other codebases the more distant the metric declaration is from its use makes it harder to be sure it was declared correctly unit errors like MB vs MBit are my bane.
  - For this case, the document and batch processing are features of this code.  Its a little messy that batches directly relate to a batch index request and I could see moving some of this processing onto the OpenSearchClient - I am OK with promoting/demoting where metrics are captured as long as we have a notation idea who is responsible.  While these metrics are easy to push downsteam, I think how often the Flux queue is blocked would need to be tracked here.  
  
Lets revisit this on your PR & future iteratinos rather than this revision.